cmd/vc-rest/startcmd/start.go

```diff
@@ -221,34 +252,57 @@ func getVCRestParameters(cmd *cobra.Command) (*vcRestParameters, error) {
 		hostURLExternal:      hostURLExternal,
 		universalResolverURL: universalResolverURL,
 		mode:                 mode,
-		databaseType:         databaseType,
-		databaseURL:          databaseURL,
+		dbParameters:         dbParams,
 		tlsSystemCertPool:    tlsSystemCertPool,
 		tlsCACerts:           tlsCACerts,
-		databasePrefix:       databasePrefix,
 	}, nil
 }
 
-func getDBConfig(cmd *cobra.Command) (string, string, string, error) {
+func getDBParameters(cmd *cobra.Command) (dbParameters, error) {
 	databaseType, err := cmdutils.GetUserSetVarFromString(cmd, databaseTypeFlagName,
 		databaseTypeEnvKey, false)
 	if err != nil {
-		return "", "", "", err
+		return dbParameters{}, err
 	}
 
 	databaseURL, err := cmdutils.GetUserSetVarFromString(cmd, databaseURLFlagName,
 		databaseURLEnvKey, true)
 	if err != nil {
-		return "", "", "", err
+		return dbParameters{}, err
 	}
 
 	databasePrefix, err := cmdutils.GetUserSetVarFromString(cmd, databasePrefixFlagName,
 		databasePrefixEnvKey, true)
 	if err != nil {
-		return "", "", "", err
+		return dbParameters{}, err
+	}
+
+	keyDatabaseType, err := cmdutils.GetUserSetVarFromString(cmd, kmsSecretsDatabaseTypeFlagName,
+		kmsSecretsDatabaseTypeEnvKey, false)
+	if err != nil {
+		return dbParameters{}, err
 	}
 
-	return databaseType, databaseURL, databasePrefix, nil
+	keyDatabaseURL, err := cmdutils.GetUserSetVarFromString(cmd, kmsSecretsDatabaseURLFlagName,
+		kmsSecretsDatabaseURLEnvKey, true)
+	if err != nil {
+		return dbParameters{}, err
+	}
+
+	keyDatabasePrefix, err := cmdutils.GetUserSetVarFromString(cmd, kmsSecretsDatabasePrefixFlagName,
+		kmsSecretsDatabasePrefixEnvKey, true)
+	if err != nil {
+		return dbParameters{}, err
+	}
+
+	return dbParameters{
```

Comment:
I usually return reference to bigger structs

Reply:
Done